### PR TITLE
Cleanup and document Container class

### PR DESCRIPTION
**Pull request by leotrs**
## List of Changes
- Documentation: added module, class, and method docstrings. @PgBiel please check formatting?
- New feature: `Container` has been made into an Abstract Base Class (`ABC`). This removes the need to raise exceptions in each abstract method. An `ABC` cannot be instantiated. Classes than inherit from an `ABC` _must_ override the abstract methods (signaled with the `@abstractmethod` decorator) before they can be instantiated.

## Motivation
- The documentation is lacking in the whole codebase, so I'm trying to start small. This file was the obvious choice.
- The `Container` class was an abstract class already, but it's always better to use the features provided by python for this purpose. Making it an `ABC` and using the decorator is not a breaking change and does not affect the user in any way (or the other parts of the codebase, for that matter).

## Testing Status
Good.

## Further Comments
See #288. I really want to move `manim/container/container.py` to just `manim/container.py`. The additional folder is not helpful in any way that I can see.

## Acknowledgement
- [ x ] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Comment by leotrs** [comment]
Latest commit will close #288

**Comment by Aathish04** [comment]
I think we can merge this as soon as @PgBiel approves.

File: manim/container.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+__all__ = ["Container"]
+
+
+class Container(ABC):
+    """Abstract base class for several manim objects.
+
+    Parameters
+    ----------
+    kwargs : :class:`dict`
```
Comment (PgBiel) [inline manim/container.py:17]: type of `kwargs` should be the type of the expected values.
So prob looking for `kwargs : Any` here (given the explanation in the description)

File: manim/container.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+__all__ = ["Container"]
+
+
+class Container(ABC):
+    """Abstract base class for several manim objects.
+
+    Parameters
+    ----------
+    kwargs : Any
+        Arguments to be passed to :func:`digest_config`
```
Comment (PgBiel) [inline manim/container.py:18]:
```suggestion
        Arguments to be passed to :func:`~.digest_config`
```

File: manim/container.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    """
+
+    def __init__(self, **kwargs):
+        digest_config(self, kwargs)
+
+    @abstractmethod
+    def add(self, *items):
+        """Abstract method to add items to Container."""
```
Comment (PgBiel) [inline manim/container.py:26]: Should prob document `items`, no? (Even if just as `Any`)

File: manim/container.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    @abstractmethod
+    def remove(self, *items):
+        """Abstract method to remove items from Container."""
```
Comment (PgBiel) [inline manim/container.py:30]: ^

File: manim/container.py (hunk 3 above)
Comment (leotrs) [reply manim/container.py]: done

File: manim/container.py (hunk 4 above)
Comment (leotrs) [reply manim/container.py]: done

**Pull request merged**